### PR TITLE
... And then there were none

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

Add tooltips to charts that mainly use [Primitive/Simple](https://formidable.com/open-source/victory/docs/victory-primitives) Victory components.

 - [x] Bullet Chart,
 - [x] Nested Area Chart,
   - [x] Nested Circle,
   - [x] Nested Square,
- [x] Comparison Bar Chart
  - [x] Reimplement using Victory simple components, and
  - [x] Add tooltip support
- [x] Test and fix tooltip support for all charts.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Peek 2019-10-28 14-42

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/core/BulletChart/index.js [hunk 1]
```diff
@@ -70,8 +70,14 @@ function BulletChart({
             reference={reference}
             style={{
               ...computedStyle,
-              data: { fill: chart.colorScale[i % chart.colorScale.length] }
+              data: {
+                fill:
+                  chart.colorScale[
+                    (computedData.length - i - 1) % chart.colorScale.length
```
Comment (karimkawambwa) [inline src/core/BulletChart/index.js]: Comment on this a bit.

File: src/core/ComparisonBarChart.js [hunk 2]
```diff
@@ -1,112 +1,151 @@
...
+        </React.Fragment>
+      ))}
+      <VictoryLabel
+        capHeight={0}
+        dy={0}
+        lineHeight={0}
+        style={referenceStyle.data}
+        text={referenceData.y}
+        x={0}
+        y={data.length * 40 + (data.length - 1) * 10 - 2 * barHeight + 70}
```
Comment (karimkawambwa) [inline src/core/ComparisonBarChart.js]: I know we use alot of magic numbers, lets note this somewhere to start explaining any numbers we use and probably have them as static variables. 

This is for future work.

File: src/core/NestedProportionalAreaChart/ScaledCircle.js [hunk 3]
```diff
@@ -35,8 +37,8 @@ function ScaledCircle({
     style: referenceStyle
   } = reference;
   const radii = data.map(d =>
-    d.x !== referenceData.x
-      ? (Math.sqrt(d.x) * size) / Math.sqrt(referenceData.x)
+    d.y !== referenceData.y
+      ? (Math.sqrt(d.y) * size) / Math.sqrt(referenceData.y)
```
Comment (karimkawambwa) [inline src/core/NestedProportionalAreaChart/ScaledCircle.js]: 😱 you mean you were using x this whole time!

File: src/core/NestedProportionalAreaChart/VerticalLegend.js [hunk 4]
```diff
@@ -43,7 +43,7 @@ function VerticalLegend({
           x={x}
           dx={0}
           y={90} // 100 - 10
-          text={formatNumberForLabel(d.x)}
+          text={formatNumberForLabel(d.y)}
```
Comment (karimkawambwa) [inline src/core/NestedProportionalAreaChart/VerticalLegend.js]: 😱 😨

File: src/core/PieChart/index.js [hunk 5]
```diff
@@ -166,14 +167,15 @@ function PieChart({
...
       {...tooltipProps}
       labelComponent={<Label colorScale={colorScale} />}
-      orientation={data2 && data2.length > 0 ? 'left' : undefined}
+      orientation={isComparisonMode ? 'left' : undefined}
       renderInPortal={false}
     />
   );
   let labelComponent2 = labelComponent1;
-  if (data2 && data2.length > 0 && !donut) {
+  if (isComparisonMode && !donut) {
```
Comment (karimkawambwa) [inline src/core/PieChart/index.js]: 👍

File: src/core/NestedProportionalAreaChart/ScaledCircle.js (hunk 3 above)
Comment (kilemensi) [reply src/core/NestedProportionalAreaChart/ScaledCircle.js]: Damn, we're shooting messengers now? I'm only _fixing_ them :face_with_head_bandage:

**Pull request merged**